virtcontainers/container.go

```diff
@@ -362,12 +363,28 @@ func (c *Container) mountSharedDirMounts(hostSharedDir, guestSharedDir string) (
 		filename := fmt.Sprintf("%s-%s-%s", c.id, hex.EncodeToString(randBytes), filepath.Base(m.Destination))
 		mountDest := filepath.Join(hostSharedDir, c.sandbox.id, filename)
 
-		if err := bindMount(m.Source, mountDest, false); err != nil {
-			return nil, err
+		var ephiVol = false
+		splitSourceSlice := strings.Split(m.Source, "/")
+		if len(splitSourceSlice) > 2 {
+			k8sStorageType := splitSourceSlice[len(splitSourceSlice)-2]
+			if k8sStorageType == "kubernetes.io~empty-dir" {
```

Comment:
Yep, this is also what I am worrying about now, it's a bit too "K8S" specific.

Reply:
Can we move the detection to kata cli to keep the knowledge out of virtcontainers? We can add a tmpfs device type for it in container DeviceInfos and kata cli is responsible for detecting ephemeral volumes and construct proper deviceinfo there.

Reply:
@WeiZhang555 any solution to handle ephemeral volumes will be very specific to the k8s. Ephemeral volumes are k8s concept after all. But we are interested in bringing them in kata because it helps to increase the isolation with the host. If you have a scenario where you need to share the data between say, k8s init and regular containers of the pod then those containers can share the data amongst themselves without explicitly putting it on the host filesystem. 

@jodh-intel ephemeral volumes are define in k8s yaml at pod level like this, 

```
apiVersion: v1
kind: Pod
metadata:
  name: myapp-pod
  annotations:
    io.kubernetes.cri.untrusted-workload: "true"
  labels:
    app: myapp
spec:
  containers:
  # Application Container
  - name: myapp-container
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    imagePullPolicy: IfNotPresent 
    image: localhost:5000/app_image:latest
    command: ['sh']
  initContainers:
  - name: data-container
    volumeMounts:
    - mountPath: /cache
      name: cache-volume
    imagePullPolicy: IfNotPresent 
    image: localhost:5000/data_image:latest
    command: ['sh', '-c', 'cp /test.gpg /cache/']
  volumes:
  - name: cache-volume
    emptyDir:
      medium: "Memory"
```

So far the SanboxConfig.Volumes (https://github.com/kata-containers/runtime/blob/master/virtcontainers/sandbox.go#L444) doens't register this volume type. 

This is the reason I resorted to parsing the OCI mounts of the invidual container mounts to infer if there are any k8s ephemeral volumes.